hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/OzoneContainer.java

```diff
@@ -478,10 +478,10 @@ public void start(String clusterId) throws IOException {
     replicationServer.start();
     datanodeDetails.setPort(Name.REPLICATION, replicationServer.getPort());
 
-    writeChannel.start();
-    readChannel.start();
     hddsDispatcher.init();
     hddsDispatcher.setClusterId(clusterId);
+    writeChannel.start();
```

Comment:
@jianghuazhu thanks for reporting the issue and find the root cause. 
If I understand it correctly, this order change is the key to fix the problem, right?  Any special reason that a ReentrantReadWriteLock is introduced in Handler.java?

Reply:
Thanks @ChenSammi  for the comment and review.
Yes, swapping the order here solves the problem.
Adding ReentrantReadWriteLock seems to be too strict, I will update it later.

Reply:
I have updated it.
@ChenSammi , can you review it again?
Thanks.